File: Data Structures & Algorithms/count-paths/submission-0.java
class Solution {
    int numberOfPath = 0;
    private void dfs(int i, int j, int m, int n){
        System.out.println(""+i+" "+j);
        if (i == m && j == n) {
            numberOfPath+=1;
            return;
        }
        if(i == m && j<n) {
            dfs(i, j+1, m, n);
        }
        else if(j == n && i<m) {
            dfs(i+1, j, m, n);
        }
        else {
            dfs(i, j+1, m, n);
            dfs(i+1, j, m, n);
        }
        
    }
    public int uniquePaths(int m, int n) {
        dfs(0, 0, m-1, n-1);
        return numberOfPath;
    }
}

File: Data Structures & Algorithms/redundant-connection/submission-0.py
class Solution:
    def findRedundantConnection(self, edges: List[List[int]]) -> List[int]:
        adj = {i:[] for i in range(1, len(edges)+1)}
        

        def dfs(curr, pre):
            # print(visit)
            if curr in visit:
                return True

            visit.add(curr)
            for child in adj[curr]:
                if child == pre:
                    continue
               
                if dfs(child, curr):
                    return True

            return False

        for u, v in edges:
            adj[u].append(v)
            adj[v].append(u)
            pre = -1
            visit = set()
            
            if dfs(u, -1):
                return [u, v]
        return []
